lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -24,6 +26,21 @@
     'lti',
 ]
 
+# exclude these in Metrics
+NON_STUDENT_ROLES = ['instructor', 'staff']
+
+
+def get_non_student_list(course_key):
+    """
+    Find all user_ids with instructor or staff roles in student_courseaccessrole table
+    """
+    non_students = CourseAccessRole.objects.select_related('user').filter(
+        course_id=course_key,
+        role__in=NON_STUDENT_ROLES,
+    ).values('user__id')
+
+    return [non_student['user__id'] for non_student in non_students]
```

Comment:
You can simplify this function to:

    non_students = CourseAccessRole.objects.filter(
        course_id=course_key,
        role__in=NON_STUDENT_ROLES,
    ).values('user_id').distinct()

    return [non_student['user_id'] for non_student in non_students]

CourseAccessRole (student_courseaccessrole) has user_id column.
The original query returned duplicate rows for members who are instructors since they have one row as staff and one row as instructor.